pkg/state/state_nv_ipam_cni_test.go

```diff
@@ -21,33 +21,389 @@ import (
+	cRole := &rbacv1.ClusterRole{}
+	err = c.Get(context.Background(), types.NamespacedName{Name: compName}, cRole)
+	Expect(err).NotTo(HaveOccurred())
+	assertClusterRole(cRole)
+	By("Verify cluster role binding")
+	cRoleBinding := &rbacv1.ClusterRoleBinding{}
+	err = c.Get(context.Background(), types.NamespacedName{Name: compName}, cRoleBinding)
+	Expect(err).NotTo(HaveOccurred())
+	expected := getExpectedNVIpamClusterRoleBinding(compName, namespace)
+	Expect(cRoleBinding.RoleRef).To(BeEquivalentTo(expected.RoleRef))
+	Expect(cRoleBinding.Subjects).To(BeEquivalentTo(expected.Subjects))
+}
+
+func assertClusterRole(clusterRole *rbacv1.ClusterRole) {
+	switch clusterRole.Name {
+	case "nv-ipam-node":
+		Expect(clusterRole.Rules).To(BeEquivalentTo(getExpectedNVIpamNodeClusterRole().Rules))
+	case "nv-ipam-controller":
+		Expect(clusterRole.Rules).To(BeEquivalentTo(getExpectedNVIpamControllerClusterRole().Rules))
+	}
+}
+
+func assertValidatingWebhookFields(adm *admv1.ValidatingWebhookConfiguration) {
+	Expect(adm.Webhooks[0].Name).To(Equal("validate-ippool.nv-ipam.nvidia.com"))
+	Expect(adm.Webhooks[1].Name).To(Equal("validate-cidrpool.nv-ipam.nvidia.com"))
+}
+
+func assertPodTemplateWebhookFields(template *v1.PodTemplateSpec) {
+	Expect(template.Spec.Containers[0].Args).Should(ContainElement(ContainSubstring("--webhook=true")))
+	Expect(template.Spec.Volumes).To(Equal(
+		[]v1.Volume{
+			{
+				Name: "cert",
+				VolumeSource: v1.VolumeSource{
+					Secret: &v1.SecretVolumeSource{
+						SecretName:  "nv-ipam-webhook-server-cert",
+						DefaultMode: ptr.To(int32(420)),
+					},
+				},
+			},
+		},
+	))
+
+	Expect(template.Spec.Containers[0].VolumeMounts).To(Equal(
+		[]v1.VolumeMount{
+			{
+				Name:      "cert",
+				MountPath: "/tmp/k8s-webhook-server/serving-certs",
+				ReadOnly:  true,
+			},
+		},
+	))
+}
+
+func getMinimalNicClusterPolicyWithNVIpam(name string) *mellanoxv1alpha1.NicClusterPolicy {
 	cr := getTestClusterPolicyWithBaseFields()
-	cr.Spec.NvIpam = &mellanoxv1alpha1.NVIPAMSpec{
-		// TODO(killianmuldoon): Test with the webhook enabled.
+	cr.Name = "nic-cluster-policy"
+
+	// add an arbitrary resource, this prevent adding defaut cpu,mem limits
+	imageSpec := addContainerResources(getTestImageSpec(), name, "5", "3")
+	nvIpamSpec := &mellanoxv1alpha1.NVIPAMSpec{
 		EnableWebhook: false,
 		ImageSpec:     *imageSpec,
 	}
-	catalog := getTestCatalog()
-	catalog.Add(state.InfoTypeStaticConfig,
-		staticconfig.NewProvider(staticconfig.StaticConfig{CniBinDirectory: "custom-cni-bin-directory"}))
+	cr.Spec.NvIpam = nvIpamSpec
+	return cr
+}
 
-	_, s, err := state.NewStateNVIPAMCNI(fake.NewClientBuilder().Build(), "../../manifests/state-nv-ipam-cni")
-	Expect(err).NotTo(HaveOccurred())
-	It("should test that manifests are rendered and fields are set correctly", func() {
-		GetManifestObjectsTest(ctx, cr, catalog, imageSpec, s)
-	})
-})
+func getExpectedNVIpamClusterRoleBinding(name, namespace string) *rbacv1.ClusterRoleBinding {
```

Comment:
I don't think we should have tests for fields that don't have any logic around them as it will make the manifests difficult to change and update. We should only have tests for fields which are templated and particularly those which have logic associated with them.

Reply:
Similare with getExpectedNVIpamNodeClusterRole and getExpectedNVIpamControllerClusterRole